src/main/java/org/broadinstitute/hellbender/tools/reference/CheckReferenceCompatibility.java

```diff
@@ -100,51 +107,59 @@ public void onTraversalStart() {
     }
 
     private void initializeSequenceDictionaryForInput(){
-        // BAMs/CRAMs
-        if(hasReads() ^ vcfPath != null){
+        if(hasReads() && vcfPath != null){
+            throw new UserException.BadInput("Both BAM and VCF specified. Tool analyzes one input at a time.");
+        }
+        else{
+            // BAMs/CRAMs
             if (hasReads()) {
                 if (readArguments.getReadPathSpecifiers().size() > 1) {
-                    throw new UserException.BadInput("Tool analyzes one BAM at a time.");
+                    throw new UserException.BadInput("Tool analyzes one reads input at a time.");
                 }
-                dictionaryToCompare = getHeaderForReads().getSequenceDictionary();
-                dictionaryPath = readArguments.getReadPathSpecifiers().get(0);
-                dictionaryName = dictionaryPath.toPath().getFileName().toString();
+                queryDictionary = getHeaderForReads().getSequenceDictionary();
+                queryDictionaryPath = readArguments.getReadPathSpecifiers().get(0);
+                queryDictionaryName = queryDictionaryPath.toPath().getFileName().toString();
+                md5sPresent = dictionaryHasMD5s(queryDictionary);
             }
             // VCFs
             else {
-                try(final FeatureDataSource<VariantContext> vcfReader = new FeatureDataSource<>(vcfPath.toString())){
+                try (final FeatureDataSource<VariantContext> vcfReader = new FeatureDataSource<>(vcfPath.toString())) {
```

Comment:
Make this explicitly check for VCF in this else statement. `vcfPath.toString()` can still throw a null pointer exception if you get here in the code without specifying either VCF OR a bam.